groovy-language-server/src/main/java/com/palantir/groovylanguageserver/GroovyTextDocumentService.java

```diff
@@ -91,7 +92,9 @@ public GroovyTextDocumentService(CompilerWrapperProvider provider) {
 
     @Override
     public CompletableFuture<List<? extends Location>> references(ReferenceParams params) {
-        throw new UnsupportedOperationException();
+        return CompletableFuture
+                .completedFuture(provider.get().findReferences(params).stream().map(symbol -> symbol.getLocation())
+                        .filter(location -> Ranges.isValid(location.getRange())).collect(Collectors.toList()));
```

Comment:
Yes in the case of generated methods and variables. There are examples in the unit tests.